CDAP-14738 copy python transform from hydrator-plugins repository

aonischuk opened the pull request:
Copy the files which are related to python transform from hydrator-plugins/core-plugins.

Some classes were shared between JavaScriptTransform and PythonTransform. 
That's why I had to move them to hydrator-plugins/hydrator-commons. That change should as well be pushed to hydrator-plugins repo for this code to fully work.

albertshau commented on widgets/PythonEvaluator-transform.json:
```diff
@@ -0,0 +1,48 @@
+  ],
+  "emit-alerts": true,
+  "emit-errors": true,
+  "errorDataset": {
```
remove this field, as error datasets are no longer supported (this shouldn't have been there in hydrator-plugins)

albertshau commented on pom.xml:
```diff
@@ -0,0 +1,600 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  Copyright © 2015-2016 Cask Data, Inc.
```
update the copyright to 2019

albertshau commented on pom.xml:
```diff
@@ -0,0 +1,600 @@
+    </dependency>
+    <dependency>
+      <groupId>co.cask.cdap</groupId>
+      <artifactId>cdap-api-spark</artifactId>
```
don't think this should be needed

albertshau commented on pom.xml:
```diff
@@ -0,0 +1,600 @@
+      <groupId>co.cask.cdap</groupId>
+      <artifactId>cdap-etl-proto</artifactId>
+      <version>${cdap.version}</version>
+      <scope>provided</scope>
```
this should only be used in tests and not in the source code, so the scope should be test

albertshau commented on pom.xml:
```diff
@@ -0,0 +1,600 @@
+    </dependency>
+    <dependency>
+      <groupId>co.cask.cdap</groupId>
+      <artifactId>cdap-etl-api-spark</artifactId>
```
don't think this is needed

albertshau commented on pom.xml:
```diff
@@ -0,0 +1,600 @@
+    </dependency>
+    <dependency>
+      <groupId>co.cask.cdap</groupId>
+      <artifactId>cdap-formats</artifactId>
```
this shouldn't be needed

albertshau commented on pom.xml:
```diff
@@ -0,0 +1,600 @@
+    </dependency>
+    <dependency>
+      <groupId>org.apache.hadoop</groupId>
+      <artifactId>hadoop-minicluster</artifactId>
```
don't think this is needed

albertshau commented on pom.xml:
```diff
@@ -0,0 +1,600 @@
+    </dependency>
+    <dependency>
+      <groupId>commons-lang</groupId>
+      <artifactId>commons-lang</artifactId>
```
Is this used in the python transform? I hope not.

albertshau commented on pom.xml:
```diff
@@ -0,0 +1,600 @@
+    </dependency>
+    <dependency>
+      <groupId>org.mockftpserver</groupId>
+      <artifactId>MockFtpServer</artifactId>
```
not needed

albertshau commented on pom.xml:
```diff
@@ -0,0 +1,600 @@
+    </dependency>
+    <dependency>
+      <groupId>commons-httpclient</groupId>
+      <artifactId>commons-httpclient</artifactId>
```
not needed

albertshau commented on pom.xml:
```diff
@@ -0,0 +1,600 @@
+    </dependency>
+    <dependency>
+      <groupId>org.apache.zookeeper</groupId>
+      <artifactId>zookeeper</artifactId>
```
not needed

albertshau commented on pom.xml:
```diff
@@ -0,0 +1,600 @@
+
+    <dependency>
+      <groupId>co.cask.hydrator</groupId>
+      <artifactId>format-common</artifactId>
```
not needed

albertshau commented on pom.xml:
```diff
@@ -0,0 +1,600 @@
+    </dependency>
+    <dependency>
+      <groupId>commons-validator</groupId>
+      <artifactId>commons-validator</artifactId>
```
not needed

albertshau commented on src/main/java/co/cask/hydrator/python/transform/PythonEvaluator.java:
```diff
@@ -0,0 +1,431 @@
+
+
+/*
+ * Copyright © 2015-2016 Cask Data, Inc.
```
update year to 2019

albertshau commented on pom.xml:
```diff
@@ -0,0 +1,600 @@
+    </dependency>
+    <dependency>
+      <groupId>org.apache.hadoop</groupId>
+      <artifactId>hadoop-common</artifactId>
```
don't think any hadoop dependency is needed

aonischuk commented:
@albertshau thanks for a detailed review. I have addressed all your comments in the follow up commit.
Also I removed every dependency from pom.xml which was possible to compile without.

Can you please re-review and if everything is okay to merge it?
